File: problem1.cpp
#include <iostream>
#include <cstdlib>

int main(int argc, char ** argv)
{
  if (argc != 2)
  {
    std::cerr << "usage: " << argv[0] << " <limit>" << std::endl;
    exit(1);
  }

  int limit = std::strtol(argv[1], nullptr, 10);

  if (limit <= 0)
  {
    std::cerr << "Limit given: " << limit << ", must be positive" << std::endl;
    exit(1);
  }

  int total = 0;

  for (int i = 0; i < limit; i += 3)
  {
    total += i;
  }

  for (int i = 0; i < limit; i += 5)
  {
    if (i % 3 != 0)
    {
      total += i;
    }
  }

  std::cout << total << std::endl;
}

File: problem1.py
#!/usr/bin/env python

import sys

def main():
  if len(sys.argv) != 2:
    print "usage: %s <limit>" % sys.argv[0]
    sys.exit(1)

  limit = int(sys.argv[1])

  total = 0

  for i in range(0, limit, 3):
    total += i

  for i in range(0, limit, 5):
    if i % 3 != 0:
      total += i

  print total

if __name__ == '__main__':
  main()
